### DISCUSSION TITLE
Define Multiple Plugins

### DISCUSSION THREAD

**Discussion by ariefsn** [nuxt/nuxt#18268] [Questions] [answered]
Hello,
want to ask something.
first time I created one plugin, and it worked.
This is displayed when typing useNuxtApp().$myPlugin
but, when I create the 2nd and 3rd plugins it doesn't show up, only my first plugin.
is it happen to you guys too?

**Comment by wobsoriano** [comment]
Can you show your code?

Example:

```ts
// plugins/your-plugin.ts
declare module '#app' {
  interface NuxtApp {
    $pluginOne: (name: string) => string;
    $pluginTwo: (age: number) => string;
  }
}

export default defineNuxtPlugin(nuxtApp => {
  nuxtApp.provide('pluginOne', (name: string) => `hello ${name}`)
  nuxtApp.provide('pluginTwo', (age: number) => `Your age is: ${age}`)
})
```

Try to delete `.nuxt` and run `yarn dev` again.

**Reply by ariefsn** [reply]
I make 3 separate files,

```
// plugins1.ts
declare module '#app' {
  interface NuxtApp {
    $plugin1: Plugin1
  }
}
```

plugins2.ts
```
// plugins2.ts
declare module '#app' {
  interface NuxtApp {
    $plugin2: Plugin2
  }
}
```

```
// plugins3.ts
declare module '#app' {
  interface NuxtApp {
    $plugin3: Plugin3
  }
}
```

or you can check my code here: https://github.com/ariefsn/nuxt3-base/tree/dev
@wobsoriano

**Reply by wobsoriano** [reply]
Try to delete `.nuxt` and run `yarn dev` again.

**Reply by ariefsn** [reply]
I did it couples of time, but same.
is there something wrong with my code?

**Reply by wobsoriano** [reply]
Your code looks fine. Try to delete both `.nuxt` and `.nuxt.d.ts` and run dev server again. Restart editor too

**Reply by ariefsn** [reply]
still not work :(

I have 3 plugins, and I try to delete it one by one, here the results, its ascending by name
1. http
2. store
3. utils

with that structure, only `$http` shown, I can access the `$store` and `$utils` but types become `any`.
then I delete the `http`, and only `$store` shown
finally I delete the `store`, and `$utils` just shown

confusing :(

**Reply by ariefsn** [reply]
the last that I try is like this.
still with 3 separate files.
on the 2 bottom files (store & utils) I only define the plugins, not declare.
and on the first files (http), I define the plugins, and declare all of plugins here.

```
// http.ts or first plugin file (asc by filename)
export default defineNuxtPlugin(nuxtApp => {
  nuxtApp.provide('http', http);
})

declare module '#app' {
  interface NuxtApp {
    $http: Http
    $store: Store
    $utils: Utils
  }
}
```

and its work for now, I dont know why.

![image](https://user-images.githubusercontent.com/33569737/138544296-99cd1169-3f4c-463e-87fd-076302b5ca35.png)

**Comment by ariefsn** [comment, marked as answer]
[temp solution](https://github.com/nuxt/framework/discussions/1386#discussioncomment-1524202)
